evalml/tests/component_tests/test_varmax_regressor.py

```diff
@@ -0,0 +1,393 @@
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=0)
+        .reset_index(drop=True)
+    )
+    X.ww["bool_2"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=1)
+        .reset_index(drop=True)
+    )
+    y = pd.DataFrame({"target_1": np.random.rand(10), "target_2": np.random.rand(10)})
+
+    vx = VARMAXRegressor(time_index="dates", series_id="series_id")
+    vx.fit(X, y)
+
+    pd.testing.assert_series_equal(
+        mock_fit.call_args[1]["X"]["bool_1"],
+        X["bool_1"].astype(float),
+    )
+    pd.testing.assert_series_equal(
+        mock_fit.call_args[1]["X"]["bool_2"],
+        X["bool_2"].astype(float),
+    )
+
+
+def test_varmax_boolean_features_no_error():
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=10)})
+    X.ww.init()
+    X.ww["bool_1"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=0)
+        .reset_index(drop=True)
+    )
+    X.ww["bool_2"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=1)
+        .reset_index(drop=True)
+    )
+    y = pd.DataFrame({"target_1": np.random.rand(10), "target_2": np.random.rand(10)})
+
+    vx = VARMAXRegressor(time_index="dates", series_id="series_id")
+    vx.fit(X, y)
+    preds = vx.predict(X)
+    assert all(preds.isna().eq(False))
+
+
+@patch("sktime.forecasting.varmax.VARMAX.fit")
+@patch("sktime.forecasting.varmax.VARMAX.predict")
+def test_varmax_regressor_respects_use_covariates(
+    mock_predict,
+    mock_fit,
+    ts_multiseries_data,
+):
+    X_train, X_test, y_train = ts_multiseries_data(n_series=2)
+    clf = VARMAXRegressor(use_covariates=False, series_id="series_id")
+
+    mock_returned = pd.DataFrame({"lower": [1] * 10, "upper": [2] * 10})
+    mock_returned = pd.concat({0.95: mock_returned}, axis=1)
+    mock_returned = pd.concat({"Coverage": mock_returned}, axis=1)
+    mock_predict.return_value = mock_returned
+
+    clf.fit(X_train, y_train)
+    clf.predict(X_test)
+    mock_fit.assert_called_once()
+    assert "X" not in mock_fit.call_args.kwargs
+    assert "y" in mock_fit.call_args.kwargs
+    mock_predict.assert_called_once()
+    assert "X" not in mock_predict.call_args.kwargs
+
+
+def test_varmax_regressor_prediction_intervals_not_implemented_yet_error(
+    ts_multiseries_data,
+):
+    X_train, X_test, y_train = ts_multiseries_data(n_series=2)
+
+    clf = VARMAXRegressor(series_id="series_id")
+
+    clf.fit(X_train, y_train)
+    with pytest.raises(
+        NotImplementedError,
+        match="VARMAX does not have prediction intervals implemented yet.",
+    ):
+        clf.get_prediction_intervals(X_test)
+
+
+@pytest.mark.parametrize(
+    "use_covariates",
+    [
+        pytest.param(
+            True,
+            marks=pytest.mark.xfail(
+                reason="Currently, using covariates with VARMAX causes inconsistent results when predicting",
```

Comment:
Do you have any guesses (for posterity/documentation) why this might be? Does this test always fail, or only fail intermittently?

Reply:
@christopherbunn is this fixable? If its not a big lift lets do it in this PR and if not let's file an issue to resolve this.

Reply:
Actually, the more that I think about it, I don't think there's really anything to fix here since we should expect a different result if we're able to feed in more past covariate data. I'm going to update this to clarify we should only expect equal results only for the case where we train the model without covariate data.